packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -0,0 +1,77 @@
+import { Element } from "hast";
+import React from "react";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+//Make sure this uses cjs, not esm. See: https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/509
+import * as style from "react-syntax-highlighter/dist/cjs/styles/prism";
+import remarkGfm from "remark-gfm";
+import { Node, Parent } from "unist";
+import { visitParents } from "unist-util-visit-parents";
+
+// Adds `inline` property to `code` elements, to distinguish between inline and block code snippets.
+function rehypeInlineCodeProperty() {
+  return function (tree: Node) {
+    visitParents(tree, "element", function (node: Node, parents: Parent[]) {
+      const element = node as Element;
+      if (element.tagName === "code") {
+        const lastParent = parents[parents.length - 1] as Element;
+        if (!element.properties) element.properties = {};
+        //It should be a string, because it can't be a boolean.
+        element.properties.inline = String(
+          !(lastParent && lastParent.tagName === "pre")
+        );
+      }
+    });
+  };
+}
+
+type MarkdownViewerProps = {
+  md: string;
+};
+
+const MarkdownViewer: React.FC<MarkdownViewerProps> = ({ md }) => {
+  return (
+    <ReactMarkdown
+      className="prose prose-stone prose-sm"
+      rehypePlugins={[rehypeInlineCodeProperty]}
+      remarkPlugins={[remarkGfm]}
+      components={{
+        pre({ children }) {
+          return <pre className="not-prose">{children}</pre>;
+        },
+        code(props) {
+          const { node, children, className, ...rest } = props;
+          const match = /language-(\w+)/.exec(className || "");
+          const isInline = node && node.properties.inline;
+          if (isInline === "true") {
+            return (
+              <code
+                {...rest}
+                className="not-prose border-black border border-opacity-[0.04] bg-opacity-[0.03] bg-black px-[0.25rem] py-0.5 rounded-sm break-words font-mono text-[.9em]"
+              >
+                {children}
+              </code>
+            );
+          }
+          return match ? (
+            <SyntaxHighlighter
```

Comment:
I think we shoud use [shiki](https://shiki.matsu.io/) instead of react-syntax-highlighter:
- that's what [Nextra uses](https://nextra.site/docs/guide/syntax-highlighting)
- it supports TextMate grammars, which we already have because that's what VS Code uses
- react-syntax-highlighter bundle size is [huge](https://bundlephobia.com/package/react-syntax-highlighter@15.5.0)

Reply:
I spent some time on this.
- I tried out  the libs "rehype-pretty-code": "^0.12.3", "rehype-shikiji": "^0.9.15",
- Neither seems to sync well with react-markdown.
<img width="1041" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/f57163ba-22d7-4dfb-b86a-39a57ef9d6ce">

I think that this change might well mean replacing react-markdown, which is too much work for now.